src/OrchardCore/OrchardCore.Abstractions/Localization/DataLocalizedString.cs

```diff
@@ -0,0 +1,17 @@
+namespace OrchardCore.Localization
+{
+    public class DataLocalizedString
+    {
+        public DataLocalizedString(string context, string name)
+        {
+            Context = context;
+            Name = name;
+        }
+
+        public string Context { get; }
```

Comment:
Context: identify the piece of data to localize. Needs to be a list of strings, example: "Customer" (the type), "Address" (the part), "Zip" (the field name)

For a content type, the Context is `["ContentType", "Name"]`.

`public string[] Context { get; }` (Key?)

Category: to group in the UI. Should it be localized?
"Content Types" -> "Types de Contenus" in the UI, so the localizer provider should have the localized string using S["Content Types"];

in DataLocalizedString:
`LocalizedString Category { get; set; }`

in ContentTypeDataLocalizer.cs we do:
return new `DataLocalizedString { Category = S["Content Types"] }`

Reply:
When the editor is displayed, the DataLocalizedString are loaded from the database. The providers are also called to get the full list of localization strings. These will be grouped by localized category. Then when displayed we use the values from the database to fill the text fields. The match is done by Context keys.

Reply:
> ContentTypeDataLocalizer has `LocalizedString LocalizedCategory {get;} => S["Content Types"]

I thought about `IDataResourceStringProvider` might have `LocalizedCategory` which is good if we want to list all the providers from DI and display them in a single UI. IMHO it would be nice if each provider uses its own view, it could be turned on or off via the features system, this way whether the above property is a member of `IDataResourceStringProvider` or not we can display the category in the UI like a normal static localized string `S["Category Name"]

Reply:
@sebastienros seems we forgot the last parameters for the `DataLocalizer` is an argument that could be used for the name/key, so context as array may not be the right option here. I'm still thinking to use something similar to message context in PO

```
msgctxt "OrchardCore.ArchiveLater.Views.ArchiveLaterPart.Edit"
```

`Customer.Address.Zip` as a context